drivers/classic/src/calc_air_temperature.c

```diff
@@ -88,34 +88,32 @@ hermint(double  xbar,
     return result;
 }
 
-/****************************************************************************/
-/*				    HourlyT                                 */
-/****************************************************************************/
-
 /******************************************************************************
- * @brief    calculate hourly temperature.
+ * @brief    calculate subdaily temperature.
  *****************************************************************************/
 void
-HourlyT(int     Dt,
-        int     ndays,
-        int    *TmaxHour,
-        double *Tmax,
-        int    *TminHour,
-        double *Tmin,
-        double *Tair)
+SubDailyT(size_t  stepsperday,
+          size_t  ndays,
+          int    *TmaxSec,
+          double *Tmax,
+          int    *TminSec,
+          double *Tmin,
+          double *Tair)
 {
     double *x;
     double *Tyc1;
     double *yc2;
     double *yc3;
     double *yc4;
-    int     i;
+    double  sec;
+    double  Dt;
+    size_t  i;
     int     j;
-    int     n;
-    int     hour;
-    int     nsteps;
+    size_t  n;
+    size_t  nsteps;
 
-    nsteps = HOURS_PER_DAY / Dt * ndays;
+    nsteps = stepsperday * ndays;
+    Dt = SEC_PER_DAY / stepsperday;
```

Comment:
Needs typecast. If not, you are rounding already.

```C
Dt = (double) SEC_PER_DAY / (double) stepsperday;
```

Also note that since `SEC_PER_DAY` is `#define`d, it is safer to do:

```C
Dt = (double) (SEC_PER_DAY) / (double) stepsperday;
```

and `SEC_PER_DAY` needs to have parentheses in the header file since it is a multiplication of two other `#define`s.